src/compiler/factory/parenthesizerRules.ts

```diff
@@ -81,6 +81,11 @@ namespace ts {
             const binaryOperatorPrecedence = getOperatorPrecedence(SyntaxKind.BinaryExpression, binaryOperator);
             const binaryOperatorAssociativity = getOperatorAssociativity(SyntaxKind.BinaryExpression, binaryOperator);
             const emittedOperand = skipPartiallyEmittedExpressions(operand);
+            // If a privateIdentifier is on the left side, this is not a real expression and
+            // should not become parenthesized
+            if (isLeftSideOfBinary && isPrivateIdentifier(operand)) {
+                return false;
+            }

```

Comment:
This shouldn't be necessary. Instead, we should add `SyntaxKind.PrivateIdentifier` to `getOperatorPrecedence` adjacent to `SyntaxKind.Identifier` and that should be sufficient:

https://github.com/microsoft/TypeScript/blob/40fa0c9f118aa1ed2b06ae0b3eed3946d5d2c612/src/compiler/utilities.ts#L3706-L3708

Reply:
yep that is a much nicer way of doing it. Thanks Ron!